lib/active_mocker/generate.rb

```diff
@@ -1,119 +1,104 @@
-                      klasses_to_be_mocked: model_names,
-                      enabled_partials:     enabled_partials,
-                      mock_append_name:     config.mock_append_name).create
-    end
-
-    OtherErrors = Struct.new(:successful?)
-    def collect_errors(mock_file_path, create_mock_errors, schema_scrapper, model_name)
-      display_errors.wrap_errors(schema_scrapper.associations.errors, model_name, type: :associations)
-      display_errors.wrap_errors(schema_scrapper.attributes.errors, model_name, type: :attributes)
-      if create_mock_errors.present? || schema_scrapper.attributes.errors.any? { |e| e.level == :error }
-        display_errors.failed_models << model_name
-        File.delete(mock_file_path) if File.exists?(mock_file_path)
-        display_errors.add(create_mock_errors)
-        OtherErrors.new(false)
-      else
-        OtherErrors.new(true)
+    attr_reader :display_errors, :progress
+
+    def check_directory!(type)
+      value = config.send(type)
+
+      if value.nil? || value.empty?
+        raise_missing_arg(type)
       end
     end
 
-    def rescue_clean_up(e, file_out, model_name)
-      display_errors.failed_models << model_name
-      file_out.close unless file_out.closed?
-      File.delete(file_out.path) if File.exists?(file_out.path)
-      display_errors.wrap_an_exception(e, model_name)
+    def raise_missing_arg(type)
+      raise ArgumentError, "#{type} is missing a valued value!"
     end
 
-    def model_name(file)
-      FilePathToRubyClass.new(base_path: config.model_dir, class_path: file).to_s
+    def write_file(model, file)
+      writer = FileWriter.new(
+        model: model,
+        file: file,
+        display_errors: display_errors,
+        config: config,
+        model_names: model_names)
+
+      writer.write!
     end
 
-    def model_names
-      @model_names ||= models_paths.map { |p| model_name(p) }
+    def progress_init
+      @progress = config.progress_class.create(active_record_models.count)
     end
 
-    def progress
-      @progress
+    def model_names
+      @model_names ||= active_record_models.map(&:name)
     end
 
-    def progress_init
-      @progress = config.progress_class.create(models_paths.count)
+    def active_record_models_with_files
+      @active_record_models_with_files ||= models_paths.map do |file|
+        model = constant_from(model_name_from(file))
+        [model, file] if model
+      end.compact
     end
 
     def models_paths
       @models_paths ||= Dir.glob(config.single_model_path || File.join(config.model_dir, "**/*.rb"))
     end
 
-    def assure_dir_path_exists(file)
-      unless File.exists?(File.dirname(file))
-        FileUtils::mkdir_p(File.dirname(file))
-      end
+    def constant_from(model_name)
+      constant = model_name.constantize
+      return unless constant.ancestors.include?(ActiveRecord::Base)
+      constant
+    rescue StandardError, LoadError => e
```

Comment:
I agree. I don't want to be a smart-ass, but you'd have to answer that yourself :) 
https://github.com/zeisler/active_mocker/blob/fec07efe1650f36c44f7b795225a4cbd4faeaa80/lib/active_mocker/generate.rb#L39

Some of the renaming I've done here was just for clarifying, and obviously makes the PR a little harder to parse, so apologies about that.

For reference: http://blog.nicksieger.com/articles/2006/09/06/rubys-exception-hierarchy/

So having NameError, LoadError instead makes sense. The previous code was just not as precise as it could have been.

I'll add that change in here, as it absolutely makes sense.